packages/deployments/contracts/contracts/core/connext/facets/SwapAdminFacet.sol

```diff
@@ -177,13 +186,51 @@ contract SwapAdminFacet is BaseConnextFacet {
     emit SwapInitialized(_key, entry, msg.sender);
   }
 
+  /**
+   * @notice remove Swap Struct for key
+   *
+   * @param _key the hash of the canonical id and domain for token
+   */
+  function removeSwap(bytes32 _key) external onlyOwnerOrAdmin {
+    uint256 numPooledTokens = s.swapStorages[_key].pooledTokens.length;
+
+    if (numPooledTokens == 0) revert SwapAdminFacet__removeSwap_notInitialized();
+
+    for (uint256 i; i < numPooledTokens; ) {
+      if (s.swapStorages[_key].balances[i] > 0) {
+        revert SwapAdminFacet__removeSwap_nonZeroBalance();
+      }
+
+      delete s.tokenIndexes[_key][address(s.swapStorages[_key].pooledTokens[i])];
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    _withdrawAdminFees(_key, msg.sender);
+
+    delete s.swapStorages[_key];
```

Comment:
if you delete this you will no longer be able to remove liquidity, even after the pool is disabled, see comments above for best approach